x-pack/plugins/monitoring/dev_docs/how_to/work_with_packages.md

```diff
@@ -0,0 +1,179 @@
+- [Package testing and validation](https://github.com/elastic/integrations/blob/main/docs/testing_and_validation.md)
+- [Developer workflow](https://github.com/elastic/integrations/blob/main/docs/developer_workflow_design_build_test_integration.md)
+
+**Related repos:**
+- [Integrations](https://github.com/elastic/integrations)
+- [Package registry](https://github.com/elastic/package-registry/)
+- [Package storage](https://github.com/elastic/package-storage/)
+- [Package spec](https://github.com/elastic/package-spec)
+
+### Integration internals
+
+#### Metrics
+For metrics collection, the integrations act as a wrapper around the corresponding metricbeat modules. For example, when the `elasticsearch` integration is configured to collect the `cluster_stats` metrics from a given host, the elastic agent will spawn a metricbeat process with the appropriate configuration.
+
+Here's the state of the running processes (configuration have been shortened for clarity):
+
+```mermaid
+
+flowchart TB
+subgraph elastic-agent
+  ESinteg["elasticsearch integration"]
+  subgraph policy.yml
+    Agentpolicy["<pre>inputs:\n  - id: elasticsearch/metrics-elasticsearch\n    type: elasticsearch/metrics\n    streams:\n      - id: elasticsearch/metrics-elasticsearch.stack_monitoring.cluster_stats\n    metricsets:\n      - cluster_stats\n    hosts:\n      - http://foo:9200\n    ...</pre>"]
+    style Agentpolicy text-align:left
+  end
+end
+
+subgraph metricbeat
+  ESmodule["elasticsearch module"]
+  subgraph metricbeat.yml
+    MBconfig["<pre>metricbeat.modules:\n  - module: elasticsearch\n    metricsets:\n      - cluster_stats\n    hosts:\n      - http://foo:9200\n    ...</pre>"]
+    style MBconfig text-align:left
+  end
+end
+
+elastic-agent-.->|spawn|metricbeat
+ESmodule-.->|config|metricbeat.yml
+ESinteg-.->|config|policy.yml
+```
+
+#### Logs
+Similarly, logs collection will spawn a filebeat process. The corresponding filebeat module, if it exists, is not used in this scenario and instead the configuration/ingest pipelines are duplicated in the integration.
+
+### Defining service dependencies
+
+A package can define the services it needs to monitor for development and automated testing purposes, for example the mysql package needs a running database to fetch and validate metrics or logs. While we could use the elasticsearch and kibana services started by `elastic-package stack up` we don't want the stack packages to be special cases and just like mysql, the packages should be responsible of defining their dependencies.
+
+This can be done by creating a `docker-compose` file under the package `_dev/deploy` directory, then running `elastic-package service up -v` in the package folder. An example is the [elasticsearch package](https://github.com/elastic/integrations/tree/main/packages/elasticsearch/_dev/deploy/docker) that starts a service which generates every types of logs with the help of a script executing queries.
```

Comment:
```suggestion
A package can define the services it needs to monitor for development and automated testing purposes, for example, the mysql package needs a running database to fetch and validate metrics or logs. While we could use the elasticsearch and kibana services started by `elastic-package stack up` we don't want the stack packages to be exceptional cases and just like mysql, the packages should be responsible for defining their dependencies.

This can be done by creating a `docker-compose` file under the package `_dev/deploy` directory and then running `elastic-package service up -v` in the package folder. An example is the [elasticsearch package](https://github.com/elastic/integrations/tree/main/packages/elasticsearch/_dev/deploy/docker) that starts a service which generates every type of log with the help of a script executing queries.
```